src/gs-category-page.c

```diff
@@ -340,12 +343,13 @@ gs_category_page_create_filter_list (GsCategoryPage *self,
 			 * "empty state" message for an empty USB disk
 			 */
 			if (g_strcmp0 (gs_category_get_id (category), "usb") == 0) {
-				/* TODO: show a "No apps available on USB"
-				 * message (after populating?) */
-
 				gs_category_page_populate_filtered (self, s);
+				category_is_usb = TRUE;
 			}
 
+			gtk_widget_set_visible (self->no_apps_box, category_is_usb);
+			gtk_widget_set_visible (self->scrolledwindow_category, !category_is_usb);
```

Comment:
Nitpick: You could bind the `visible` properties with the invert option. But we can merge this as is as I know you're in a rush.